pandas/core/groupby/generic.py

```diff
@@ -1475,6 +1481,11 @@ def _transform_fast(self, result: DataFrame, func_nm: str) -> DataFrame:
         # for each col, reshape to to size of original frame
         # by take operation
         ids, _, ngroup = self.grouper.group_info
+
+        # in categorical case there may be unobserved categories in index
```

Comment:
does this depend on if observed=True is passed?

Reply:
It's unnecessary if `observed` is True - have taken care of that now, thanks

Reply:
can we *always just do
``result = result.reindex(self.groupber.result_index, copy=False)`` ?

what breaks if we do that

Reply:
@jreback that's fine, we can do that always, I just thought it would be better not to if it wasn't necessary. Green now